fog/ingest/client/src/main.rs

```diff
@@ -168,6 +180,42 @@ fn sync_keys_from_remote(
     Ok(())
 }
 
+fn get_ingress_key_records(
+    logger: &Logger,
+    ingest_client: &FogIngestGrpcClient,
+    start_block_at_least: u64,
+    should_include_lost_keys: bool,
+    should_include_retired_keys: bool,
+) -> ClientResult<()> {
+    let ingress_key_records = ingest_client
+        .get_ingress_key_records(
+            start_block_at_least,
+            should_include_lost_keys,
+            should_include_retired_keys,
+        )
+        .expect("Failed getting ingress key records");
+
+    log::info!(logger, "Ingress keys successfully retrieved");
+    println!(
+        "{}",
+        json!(ingress_key_records
+            .iter()
+            .map(|record| {
+                json!({
+                    "ingress_public_key": hex::encode(record.get_ingress_public_key().get_data()),
+                    "start_block": record.start_block,
+                    "pubkey_expiry": record.pubkey_expiry,
+                    "retired": record.retired,
+                    "lost": record.lost,
+                    "last_scanned_block": record.last_scanned_block
```

Comment:
This "last_scanned_block" value comes from this [struct](https://github.com/mobilecoinfoundation/mobilecoin/blob/bcb3e26cc85afd3a250f2109e9cd388c0e5c814a/fog/recovery_db_iface/src/types.rs#L61)'s last_scanned_block field, which is an Option<u64>. I chose to not set the value on the proto if it's not set (see comment below), and I believe that because we're using proto3, it will just set the field to the uint64 default value of 0.

I believe that I could check to see if this field is set using something like record.has_last_scanned_block (?), and if it doesn't have a set value, I could use a string value like "UNKNOWN" instead of just passing the default proto uint64 value of 0. Let me know if you'd like me to investigate this further.

Reply:
I think this is fine as is since fog did not scan block index 0.